File: WebsiteHoTroDichVuBanDoc_Backend/WebsiteHoTroDichVuBanDoc_Backend/app/models/loai_the.py
from decimal import Decimal
from typing import Optional
from app.models.db_base import DBModel

class LoaiTheBase(DBModel):
    tenThe: str
    moTa: Optional[str] = None
    taiLieuMuonToiDa: int = 3
    soNgayMuonMacDinh: int = 14
    lePhi: Optional[Decimal] = 0

class LoaiTheCreate(LoaiTheBase):
    pass

class LoaiTheUpdate(DBModel):
    tenThe: Optional[str] = None
    moTa: Optional[str] = None
    taiLieuMuonToiDa: Optional[int] = None
    soNgayMuonMacDinh: Optional[int] = None
    lePhi: Optional[Decimal] = None

class LoaiThe(LoaiTheBase):
    maLoaiThe: int

File: WebsiteHoTroDichVuBanDoc/Backend/src/main/java/com/library/website/models/LoaiThe.java
package com.library.website.models;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "loaithe", schema = "public")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class LoaiThe {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "maloaithe")
    private Short maLoaiThe;

    @Column(name = "tenthe", unique = true, nullable = false)
    private String tenThe;

    @Column(name = "mota")
    private String moTa;

    @Column(name = "tailieumuontoida")
    private Integer taiLieuMuonToiDa = 3;

    @Column(name = "songaymuonmacdinh")
    private Integer soNgayMuonMacDinh = 14;

    @Column(name = "lephi")
    private BigDecimal lePhi = BigDecimal.ZERO;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt = LocalDateTime.now();

    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}
